common/edgecentral.go

```diff
@@ -38,8 +38,12 @@ func ParseValueByDeviceResource(valueType string, value interface{}) (interface{
 		var arr []string
 		err = json.Unmarshal([]byte(v), &arr)
 		if err != nil {
-			errMsg := fmt.Sprintf("failed to convert set parameter %s to ValueType %s", v, valueType)
-			return nil, errors.NewCommonEdgeX(errors.KindServerError, errMsg, err)
+			// try to handle the nonstandard json format, for example, [foo, bar]
+			strArr := strings.Split(strings.Trim(v, "[]"), ",")
+			for _, u := range strArr {
+				arr = append(arr, strings.Trim(u, " "))
```

Comment:
```suggestion
				arr = append(arr, strings.TrimSpace(u))
```
If the pupose is only to trim the spaces, use TrimSpace instead

Reply:
Modified.